lib/liquid/document.rb

```diff
@@ -19,16 +19,15 @@ def nodelist
       @body.nodelist
     end
 
-    def parse(tokens, parse_context)
-      @body.parse(tokens, parse_context) do |end_tag_name, _end_tag_params|
-        unknown_tag(end_tag_name, parse_context) if end_tag_name
+    def parse(tokenizer, parse_context)
+      while parse_body(tokenizer)
```

Comment:
Why are we adding a loop?

Reply:
Because we don't want the handled unknown tag (e.g. section tags) to end the parsing of the document, we still need to parse the rest of the document.

Reply:
Right, but why wasn't there a loop in the previous implementation? Sorry if I'm missing something obvious 😅 I reviewed expecting a refactor without change in behavior.

Reply:
Previously `unknown_tag` was expected to always raise when called on a `Liquid::Document`, unlike `Liquid::Block` which has its `unknown_tag` method overridden by child classes which don't always raise.  This won't change any existing code, since `Liquid::Document#unknown_tag` itself still always raises.  It just means that now we can override `unknown_tag` in a child class of `Liquid::Document` and not raise to have it continue parsing.

Reply:
>  It just means that now we can override `unknown_tag` in a child class of `Liquid::Document` and not raise to have it continue parsing.

Ahhh that explains it. 👍